### PR TITLE
fix(components):[select] placeholder local

### PR DESCRIPTION
**Pull request by gjfei**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/select/src/select.vue [hunk 1]
```diff
@@ -513,16 +514,25 @@ export default defineComponent({
...
+          Array.isArray(props.modelValue) &&
+          props.modelValue.length > 0
+        ) {
+          currentPlaceholder.value = ''
+        }
+      },
+      {
+        immediate: true,
       }
+    )
```
Comment (JeremyWuuuuu) [inline packages/components/select/src/select.vue:533]: **Suggestion:** put this chunk of code inside `onMounted`

**Review by jw-foss — CHANGES_REQUESTED** [review]
Left a comment for you to consider 🙇

File: packages/components/select/src/select.vue (hunk 1 above)
Comment (gjfei) [reply packages/components/select/src/select.vue:533]: > **Suggestion:** put this chunk of code inside `onMounted`

可以告诉我原因吗？

Comment (JeremyWuuuuu) [reply packages/components/select/src/select.vue:533]: No specific reason, we should keep the change minimal. 

If you think the watch statement is ok to be put outside `onMounted` hook, you can make another PR for that.

Does this make any sense?

Comment (gjfei) [reply packages/components/select/src/select.vue:533]: 我好像不是很能理解watch放在onMounted里面，如果是用工具去对比变化的话，并没有过多的更改
image

Comment (JeremyWuuuuu) [reply packages/components/select/src/select.vue:533]: If you insist, would you mind adding a test case against your change?

**Pull request closed without merging**